### PR TITLE
Pass annotation files to last-dotplot

### PR DESCRIPTION
**Pull request by charles-plessy**
This adds the capacity to overlay annotations to the dot plots.  In the _pairgenomealign_ pipeline, we will use it to indicate genomic regions containing N stretches that typically mark contig boundaries in scaffold assemblies, like the pink lines in the following example:

image

## Details

The annotation of the target (plotted horizontally) genome is passed as a new channel.

The annotation of the query (plotted vertically) genome is passed as a new component of the first channel.

This is because this module typically loops on a list of alignments of various query genomes (from samplesheet) to a single target genome.

The annotation files are called `annot_a` and `annot_b` because they are passed to the `-a` and `-b` arguments of `last-dotplot`.

## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [ x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/last/dotplot/tests/main.nf.test
Comment (SPPearce) [inline modules/nf-core/last/dotplot/tests/main.nf.test:1]: Could you add a test with one or both of the annotations?
You can make small text files on the fly e.g. https://github.com/nf-core/modules/blob/master/modules/nf-core/taxpasta/merge/tests/main.nf.test#L67-L72

File: modules/nf-core/last/dotplot/main.nf [hunk 1]
```diff
@@ -22,9 +23,12 @@ process LAST_DOTPLOT {
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+    def annot_a_arg = annot_a ? "-a ${annot_a}" : ''
+    def annot_b_arg = annot_b ? "-b ${annot_b}" : ''
     """
     last-dotplot \\
         $args \\
+        $annot_a_arg $annot_b_arg \\
```
Comment (SPPearce) [inline modules/nf-core/last/dotplot/main.nf:31]: Can you put them onto separate lines please.

File: modules/nf-core/last/dotplot/main.nf [hunk 2]
```diff
@@ -8,8 +8,9 @@ process LAST_DOTPLOT {
         'biocontainers/last:1542--h43eeafb_1' }"
 
     input:
-    tuple val(meta), path(maf)
+    tuple val(meta), path(maf), path(annot_b)
     val(format)
+    path(annot_a)
```
Comment (SPPearce) [inline modules/nf-core/last/dotplot/main.nf:13]: I feel this probably wants a meta map, and I'd put it above the val(format) channel.

File: modules/nf-core/last/dotplot/main.nf (hunk 1 above)
Comment (charles-plessy) [reply modules/nf-core/last/dotplot/main.nf]: Done in 0166d15b22e93d568c86623c21288326e0a629d4

File: modules/nf-core/last/dotplot/main.nf (hunk 2 above)
Comment (charles-plessy) [reply modules/nf-core/last/dotplot/main.nf]: Done in ad87c1c3cf0ae7ccd01b7b18cf1c6f3d06e92b9d

File: modules/nf-core/last/dotplot/tests/main.nf.test
Comment (charles-plessy) [reply modules/nf-core/last/dotplot/tests/main.nf.test:1]: Done in f29dc353849e4917c9b6772468ad79f8e3a78e1d

Thanks for the hint!

**Review by SPPearce — APPROVED** [review]
Looks good to me. 
By the way, you don't need to say which commit addresses each comment (not a problem to do so, but I check the files again anyway)

**Comment by charles-plessy** [comment]
Good to know, thank you.  Do I need to mark the discussion resolved or is it enough that the file was changed in a newer commit ?

**Comment by SPPearce** [comment]
> Good to know, thank you. Do I need to mark the discussion resolved or is it enough that the file was changed in a newer commit ?

As an author, I generally mark the discussions as resolved once I have addressed them, assuming that they are just minor fixes like these. If I want to continue the discussion then I don't.

**Pull request merged**